sparse/utils.py

```diff
@@ -240,3 +242,44 @@ def normalize_axis(axis, ndim):
         return tuple(normalize_axis(a, ndim) for a in axis)
 
     raise ValueError("axis %s not understood" % axis)
+
+
+def equivalent(x, y):
+    return (x == y) | ((x != x) & (y != y))
+
+
+def check_fill_value(nargs):
+    def generator(func):
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+            for arg in args[:nargs]:
+                if hasattr(arg, 'fill_value') and \
+                        not equivalent(arg.fill_value, _zero_of_dtype(arg.dtype)):
+                    raise ValueError('Operation {!r} is only supported with zero fill-values.'.format(func))
+
+            return func(*args, **kwargs)
+
+        return wrapped
+
+    return generator
+
+
+def consistent_fill_value(func):
+    from .sparse_array import SparseArray
+
+    @functools.wraps(func)
+    def wrapped(arrays, *args, **kwargs):
```

Comment:
As above, consider using a function instead of a decorator here.

In this case, the decorator would work OK (as long as the first argument is always called `arrays`) but decorators are more magical than simple function calls.